src/pudl/output/ferc714.py

```diff
@@ -0,0 +1,400 @@
+        priority (str): Which type of entity should take priority in the categorization
+            of FERC 714 respondents. Must be either ``utility`` or
+            ``balancing_authority.`` The default is ``balancing_authority``.
+        limit_by_state (bool): Whether to limit respondent service territories to the
+            states where they have documented activity in the EIA 861. Currently this
+            is only implemented for Balancing Authorities.
+
+    """
+
+    def __init__(
+        self,
+        pudl_out,
+        pudl_settings=None,
+        ba_ids=None,
+        util_ids=None,
+        priority="balancing_authority",
+        limit_by_state=True,
+    ):
+        """Set respondent compilation parameters."""
+        self.pudl_out = pudl_out
+
+        if pudl_settings is None:
+            pudl_settings = pudl.workspace.setup.get_defaults()
+        self.pudl_settings = pudl_settings
+
+        if ba_ids is None:
+            ba_ids = (
+                self.pudl_out.balancing_authority_eia861()
+                .balancing_authority_id_eia.dropna().unique()
+            )
+        self.ba_ids = ba_ids
+
+        if util_ids is None:
+            util_ids = get_all_utils(self.pudl_out).utility_id_eia.dropna().unique()
+        self.util_ids = util_ids
+
+        self.priority = priority
+        self.limit_by_state = limit_by_state
+        self._categorized = None
+        self._annualized = None
+        self._demand_summary = None
+        self._fipsified = None
+        self._counties_gdf = None
+        self._respondents_gdf = None
+
+    def annualize(self, update=False):
+        """
+        Broadcast respondent data across all years with reported demand.
+
+        The FERC 714 Respondent IDs and names are reported in their own table,
+        without any refence to individual years, but much of the information we are
+        associating with them varies annually. This method creates an annualized
+        version of the respondent table, with each respondent having an entry
+        corresponding to every year in which hourly demand was reported in the FERC 714
+        dataset as a whole -- this necessarily means that many of the respondents will
+        end up having entries for years in which they reported no demand, and that's
+        fine.  They can be filtered later.
+        """
+        if update or self._annualized is None:
+            # Calculate the total demand per respondent, per year:
+            report_dates = (
+                self.pudl_out.demand_hourly_pa_ferc714()
+                .report_date.unique()
+            )
+            self._annualized = (
+                self.pudl_out.respondent_id_ferc714()
+                .pipe(add_dates, report_dates)
+                .pipe(
+                    pudl.helpers.convert_cols_dtypes,
+                    data_source="ferc714",
+                    name="FERC 714 Respondents Annualized",
+                )
+            )
```

Comment:
Given the different ways that the FERC 714 and EIA 861 treat the entities responding to them, I don't know that there's a completely self-consistent way to deal with this.  Did you have something in particular in mind?

Reply:
I think unless we want to make the FERC respondent to EIA BA/Util mapping also something that varies annually, and separately attempt to track down what entities each ID should correspond to in each year (which, tbh, might be the more right thing to do, but also adds a lot more complexity to the process) then I think some kind of broadcasting like this has to be done, since other data (the service territories) certainly do vary year by year.